doc/lsst.pex.config/field-types.rst

```diff
@@ -0,0 +1,62 @@
+.. py:currentmodule:: lsst.pex.config
+
+#############################
+Types of configuration fields
+#############################
+
+Attributes of the configuration object must be subclasses of `Field`.
+A number of these are predefined: `Field`, `RangeField`, `ChoiceField`, `ListField`, `ConfigField`, `ConfigChoiceField`, `RegistryField` and `ConfigurableField`.
+
+Example of `RangeField`::
+
+    class BackgroundConfig(pexConfig.Config):
+        """Parameters for controlling background estimation."""
+        binSize = pexConfig.RangeField(
+            doc="how large a region of the sky should be used for each background point",
+            dtype=int, default=256, min=10
+        )
+
+Example of `ListField` and `Config` inheritance::
+
+    class OutlierRejectedCoaddConfig(CoaddTask.ConfigClass):
+        """Additional parameters for outlier-rejected coadds."""
+        subregionSize = pexConfig.ListField(
+            dtype = int,
+            doc = """width, height of stack subregion size; make small enough that a full stack of images will fit into memory at once""",
+            length = 2,
+            default = (200, 200),
+            optional = None,
+        )
+
+Examples of `ChoiceField` and `ConfigField` and the use of the `Config` object's `Config.setDefaults` and `Config.validate` methods::
+
+    class InitialPsfConfig(pexConfig.Config):
+        """Describes the initial PSF used for detection and measurement before
+        we do PSF determination."""
+
+        model = pexConfig.ChoiceField(
+            dtype = str,
+            doc = "PSF model type",
+            default = "SingleGaussian",
+            allowed = {
+                "SingleGaussian": "Single Gaussian model",
+                "DoubleGaussian": "Double Gaussian model",
+            },
+        )
```

Comment:
We are inconsistent about whether to put spaces around `=` in the keyword arguments here in the codebase, but I believe we have settled on not adding spaces there in the style guide.  We should make sure all of the examples here follow that.